pyomo/core/base/indexed_component.py

```diff
@@ -666,6 +666,27 @@ def _processUnhashableIndex(self, idx):
                 fixed[i - len(idx)] = val
 
         if sliced or ellipsis is not None:
+            if self.dim() is None:
+                # Assume that the right thing to do here is return
+                # an IndexedComponent_slice
```

Comment:
Huh? What case would cause this?

Reply:
I think some uninstantiated sets in abstract models can have dimension None? It shows up in one of the tests, which is why I added it.

Reply:
Also sets with inconsistent dimension can be created with dimension None. From `test_reference.py`:
```
        m.js = Set(initialize=[1, (2,3)], dimen=None)
```